hazelcast/src/main/java/com/hazelcast/concurrent/lock/operations/BeforeAwaitOperation.java

```diff
@@ -80,7 +79,8 @@ public Operation getBackupOperation() {
 
     @Override
     public WaitNotifyKey getNotifiedKey() {
-        return new LockWaitNotifyKey(namespace, key);
+        LockStoreImpl lockStore = getLockStore();
+        return lockStore.getNotifiedKey(key);
```

Comment:
we delegate to a lock store - if there is a pending signal then we have to use `ConditionKey` as a notifier.